Ларькин П.В.

desile opened the pull request.

desile committed:
draft version of cards

Commit:
Added draft portfolio and clients blocks

Commit:
BAM styled less

Commit:
Citation and team blocks

Commit:
Facts block

Commit:
Added png and margin to team block

Commit:
Merge branch 'citation_team'

Commit:
News block

Commit:
Other News block

Commit:
Added comment block

Commit:
Added footer

desile committed:
draft version of cards

Commit:
Merge branch 'master' into featured_articles

# Conflicts:
#	index.html
#	less/all.less
#	less/text.less
#	less/title.less

Commit:
End of working with articles

Commit:
fix favicon bug

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                    <div>
+                                        <span class="articles__person-card__img"></span>
+                                        <div class="articles__person-info">
+                                            <p class="articles__person-name">ANNIE MALKOVICH</p>
```
Семантически это не похоже на `<p>`. Я бы лучше сделал `<div>`.

desile replied:
Это потому что я `<p>` использую как блочный элемент?

sergdenisov replied:
Ну да. `<p>` — это параграф с текстом. Тут у тебя имя в нем лежит.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                        <span class="articles__person-card__img"></span>
+                                        <div class="articles__person-info">
+                                            <p class="articles__person-name">ANNIE MALKOVICH</p>
+                                            <p class="articles__person-date">2017/10/16</p>
```
И это.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <h2 class="title title_level-2">Clients</h2>
+                            </div>
+                        </div>
+                        <div>
```
Лишний элемент.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </div>
+                        <div>
+                            <div class="row justify-content-center">
```
Тут лучше положить все в одну строку, ту, что выше. Поскольку логически это один кусок.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <span class="img clients__img"></span>
+                                <span class="img clients__img"></span>
+                                <span class="img clients__img"></span>
+                                <div class="fa fa-angle-right fa-4x arrow clients__arrow"></div>
```
Не не не, использование стронних либ для этого задания — хак =). Это можно просто сделать нативными средствами.

desile replied:
Под нативными средствами имеется ввиду использование картинки-бэкграунда? Я наоборот думал разнообразить спрайты стильными иконками :)

sergdenisov replied:
Не предполагалось использование сторонних либ для этого. Можно либ [HTML Entity](https://dev.w3.org/html5/html-author/charref), либо картинку.

sergdenisov commented on index.html:
```diff
@@ -56,7 +57,7 @@ <h1 class="title">Storyteller</h1>
                 <div class="row justify-content-center">
                     <div class="col-10">
-                        <div class="intro">
+                        <div class="header">
```
Логически этот блок не похож на заголовок, нужно другое название, более говорящее.

desile replied:
Я оставлю название стиля header и буду оборачивать в него заголовки разделов. Такое решение подойдет?

sergdenisov replied:
Ну, наверное, нужно код смотреть))).

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </div>
+                    </div>
+                    <hr class="underline portfolio__underline">
```
`<hr>` здесь не нужен, т.к. не несет никакой семантической нагрузки.

desile replied:
Почему? `<hr>` семантически же это линия/черта.

sergdenisov replied:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr

> The **HTML `<hr>` element** represents a thematic break between paragraph-level elements (for example, a change of scene in a story, or a shift of topic with a section); historically, this has been presented as a horizontal rule or line. While it may still be displayed as a horizontal rule in visual browsers, this element is now defined in semantic terms, rather than presentational terms, so if you wish to draw a horizontal line, you should do so using appropriate CSS.

desile replied:
О как! Спасибо)

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        </div>
+                        <div>
+                            <div class="row justify-content-center">
+                                <span class="img clients__img"></span>
```
Семантически, мне кажется, тут `<ul>` — `<li>` хорошо подходят.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <span class="circle"></span>
+                        <span class="circle circle_active"></span>
+                        <span class="circle"></span>
+                    </div>
```
Все 3 блока выше семантически `<ul>` — `<li>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+            </section>
+
+            <!--CITATION-->
+            <section class="dark_substrate">
```
По БЭМу не может быть модификатора без блока/элемента.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <div class="row justify-content-center">
+                            <div class="col-8">
+                                <div class="header citation__header">
+                                    <div class="fa fa-quote-right"></div>
```
Для этого кейса тоже лучше обойтись нативными средствами.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        </div>
+                        <div class="row justify-content-center">
+                            <div class="col-8">
+                                <p class="text text_level-1">There's a special rung in hell reserved for people who waste good scotch.</p>
```
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/blockquote

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </div>
+                        <div class="row justify-content-center">
+                            <p class="text text_level-3 text_italic">Lt. Archie Hicox</p>
```
Это тоже не похоже на `<p>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <p class="text text_level-1">There's a special rung in hell reserved for people who waste good scotch.</p>
+                            </div>
+                        </div>
+                        <div class="row justify-content-center">
```
Опять же, здесь лучше чтобы все лежало в одной строке, поскольку это один блок.

desile replied:
Почему? Я всегда думал, что то что визуально выглядит как горизонтальный блок - это строка. Несколько горизонтальных блоков - несколько строк.

sergdenisov replied:
Не совсем. В данном случае, у тебя обертка каждого блочного элемента в строку + колонку не несет никакого смысла, но добавляет лишнюю вложенность. Тут достаточно положить одну строку, в нее колонку, а все элементы внутри просто должны быть блочными элементами в вертикальным отступом. Это проще.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </div>
+                    </div>
+                    <div class="row justify-content-center">
```
Тоже не нужна лишняя строка.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <div class="team__member__role">
+                                Sale Manager
+                            </div>
+                        </div>
```
Структура тоже выглядит как список.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                    </div>
+                    <div class="row justify-content-center">
+                        <div class="team__member">
+                            <div class="team__member__img1"></div>
```
Я думаю тут лучше через `<img>`, поскольку это обычно динамический контент.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                    <div class="row justify-content-center">
+                        <div class="team__member">
+                            <div class="team__member__img1"></div>
+                            <div class="team__member__name">
```
Это должно быть каким-нибудь `<h4>` — `<h6>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+
+            <!--FACTS-->
+            <div class="facts">
+                <section class="container">
```
`<section>` по идее должен быть родительским в этом случае.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                        <div class="row">
+                                            <div class="col-6">
+                                                <div class="facts__block__number">100</div>
+                                                <hr class="underline facts__block__underline">
```
Эта штука тоже чисто визуальная, я бы не стал отражать ее в структуре через `<hr>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                            </div>
+                                        </div>
+                                    </div>
+                                    <hr class="underline facts__underline">
```
И эта.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+
+            <!--NEWS-->
+            <div class="news">
+                <section class="container">
```
Тоже секцию лучше выше ставить.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <section class="container">
+                    <div class="row justify-content-center news__row">
+                        <div class="col-6 news__row__block">
+                            <div>
```
Лишний элемент.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <div class="col-6 news__row__block">
+                            <div>
+                                <div class="text text_level-3">Travel / 21st March, 2014</div>
+                                <div class="title title_level-3">A  New Product Is Coming</div>
```
`<h4>` — `<h6>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <div>
+                                <div class="text text_level-3">Travel / 21st March, 2014</div>
+                                <div class="title title_level-3">A  New Product Is Coming</div>
+                                <div class="text text_level-3">The office assistant was the boss's man, spineless, and with no understanding. What about if he reported sick? But that would be extremely strained and suspicious as in fifteen years of service Gregor had ever.</div>
```
`<p>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </div>
+                        <div class="col-6">
+                                <div class="news__row__block__img1"></div>
```
Тоже лучше через `<img>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <div class="col-4 news__other__block">
+                                    <div class="news__other__block__valign"></div>
+                                    <div>
+                                        <div class="title title_level-4">Another Day In Paradise</div>
```
Семантически — заголовок. Скорее всего, даже со ссылкой.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <div class="fa fa-twitter fa-2x"></div>
+                        </div>
+                        <div class="row justify-content-center">
+                            <div class="img comment__img"></div>
```
Тоже лучше через `<img>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <section class="container">
+                    <div class="comment">
+                        <div class="row justify-content-center">
+                            <div class="fa fa-twitter fa-2x"></div>
```
Лучше нативными средствами тоже.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <p class="text text_level-1">Chaos is found in greatest abundance wherever order is being sought. It always defeats order, because it is better organized.</p>
+                            </div>
+                        </div>
+                        <div class="row justify-content-center">
```
Лучше, чтобы все было в одной строке тоже.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+            <div class="container">
+                <div class="row title__footer">
+                    <div class="col-6">
+                        <p class="title title_level-3">UNICORN</p>
```
Тут должна быть вертикальная структура, а не горизонтальная. То есть это заголовок, а внизу список со ссылками.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                </div>
+                <div class="row justify-content-center main__footer">
+                    <div class="col-6">
+                        <div class="msg__footer">
```
Нет основного блока `msg`, без него нельзя использовать элементы типа `msg__footer`.

desile replied:
Основным блоком тут является `footer` (как например на 457 строке `title__footer` или на 471 `main__footer`)

sergdenisov replied:
В терминах БЭМа это не так. Название элемента формируется как `название-блока__название элемента`. https://ru.bem.info/methodology/naming-convention/#%D0%98%D0%BC%D1%8F-%D1%8D%D0%BB%D0%B5%D0%BC%D0%B5%D0%BD%D1%82%D0%B0

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <p class="text text_level-3">Chaos is found in greatest abundance wherever order is being sought. It always defeats order, because it is better organized.</p>
+                            <div class="row justify-content-center">
+                                <div class="col-6">
+                                    <input class="input__msg__footer" placeholder="EMAIL" type="text">
```
Нет родительского элемента `<form>` выше, без него это не запоститься на сервер.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </div>
+                    </div>
+                    <div class="col-2">
```
Это список со ссылками, остальные рядом тоже.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                    <input class="input__msg__footer" placeholder="EMAIL" type="text">
+                                </div>
+                                <div class="col-6">
+                                    <input class="input__msg__footer" placeholder="SUBJECT" type="text">
```
Лучше стилизовать заглавные буквы через CSS, а не в HTML.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                </div>
+                <div class="row">
+                    <div class="col-12">
+                        <a href="#" class="link link__footer">SEND ▶</a>
```
Это должно быть кнопкой, по клику на которую происходит пост на сервер.

sergdenisov commented on index.html:
```diff
@@ -110,8 +111,430 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <div class="row">
+                    <div class="col-12">
+                        <a href="#" class="link link__footer">SEND ▶</a>
+                        <hr class="underline footer__underline">
```
Тоже не нужен `<hr>`.

sergdenisov commented on less/all.less:
```diff
@@ -1,11 +1,42 @@
+@import "variables";
```
Это должно быть после `@import "defaults";`, иначе могут неправльно наложиться стили.

sergdenisov commented on less/all.less:
```diff
@@ -1,11 +1,42 @@
+@import "news";
+@import "footer";
+
+.comment {
```
Каждый блок должен быть в отдельном файле, в этом файле должны быть только импорты.

sergdenisov commented on less/all.less:
```diff
@@ -1,11 +1,42 @@
+    padding-bottom: 50px;
+}
+
+.clients {
```
Блоки без стилей не нужны.

sergdenisov commented on less/arrow.less:
```diff
@@ -0,0 +1,13 @@
+.arrow {
+  position: absolute;
```
Лучше не делать таким образом блок, поскольку он вряд ли может быть использован независимо.

desile replied:
Это у меня все БЭМ никак не уложится) А как тогда лучше сделать - пустым он быть не может, но у него же есть специфичные стили arrow__clients и arrow__news, они могут существовать, только если у нас существует стиль arrow?

desile replied:
В обычном случае я бы вообще убрал arrow и position: absolute перенес в специфичные стили и использовал только их. Но как я понимаю БЭМ такого не позволяет?

sergdenisov replied:
Смотри, на мой взгляд лучше делать так:
1. Блок `.arrow` содержит только стили для самой стрелки без позиционирования.
2. Там, где нужно расположить стрелку абсолютом или как-то еще, можно сделать элемент блока, типа `.clients__arrow`, который уже будет управлять позиционированием, и замиксовать с основном блоком `.arrow` или обернуть в него.

sergdenisov commented on less/arrow.less:
```diff
@@ -0,0 +1,13 @@
+  position: absolute;
+}
+
+    .clients__arrow {
```
В этом файле блок `.arrow`, поэтому тут могут быть только его элементы, типа `.arrow__...`.

sergdenisov commented on less/arrow.less:
```diff
@@ -0,0 +1,13 @@
+
+    .clients__arrow {
+        margin-top: 20px;
+        margin-left: 460px;
```
Такие штуки должны лежать скорее абсолютно спозиционированные по правой стороне контейнера, чтобы не завязываться на конкретные числа.

sergdenisov commented on less/arrow.less:
```diff
@@ -0,0 +1,13 @@
+    .news__arrow {
+        margin-left: 560px;
+        margin-top: 8px;
+    }
```
В каждом файле должны быть последняя пустая строка https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file.

sergdenisov commented on less/articles.less:
```diff
@@ -0,0 +1,45 @@
+.articles{
```
Надо отформатировать файлы нормально, чтобы ничего не слипалось.

sergdenisov commented on less/articles.less:
```diff
@@ -0,0 +1,45 @@
+}
+
+
+    .articles__person-card:hover:before{
```
Не принципиально, но удобнее читать код, когда такая штука лежит ниже элемента, к которому она относится.

sergdenisov commented on less/articles.less:
```diff
@@ -0,0 +1,45 @@
+      padding: 30px;
+    }
+
+    .articles__person-card__description{
```
В БЭМе нельзя элементы элементов создавать, я вроде говорил об этом на лекции https://ru.bem.info/methodology/faq/#%D0%9F%D0%BE%D1%87%D0%B5%D0%BC%D1%83-%D0%BD%D0%B5-%D1%81%D1%82%D0%BE%D0%B8%D1%82-%D1%81%D0%BE%D0%B7%D0%B4%D0%B0%D0%B2%D0%B0%D1%82%D1%8C-%D1%8D%D0%BB%D0%B5%D0%BC%D0%B5%D0%BD%D1%82%D1%8B-%D1%8D%D0%BB%D0%B5%D0%BC%D0%B5%D0%BD%D1%82%D0%BE%D0%B2-block__elem1__elem2. По идее, `person-card` должен стать отдельным блоком.

desile replied:
А вот это боль :) у меня такое везде и повсюду. Что-то я не так понял/прочел.

sergdenisov replied:
Ну, тут достаточно просто прочитать документацию по БЭМу, она же маленькая =).

sergdenisov commented on less/articles.less:
```diff
@@ -0,0 +1,45 @@
+      margin-bottom: 30px;
+    }
+
+    .articles__person-info{
```
Это тоже выглядит как элемент элемента. У тебя либо не должно быть каждый раз `person` в названии, либо `person` делать отдельным блоком.

sergdenisov commented on less/facts.less:
```diff
@@ -0,0 +1,15 @@
+        padding: 32px;
+    }
+
+    .facts__block__number {
```
Тоже элемент элемента.

sergdenisov commented on less/facts.less:
```diff
@@ -0,0 +1,15 @@
+    color: #f5f5f5;
+}
+
+    .facts__block {
```
Нужно выбрать другое название, это какое-то не говорящее =).

sergdenisov commented on less/footer.less:
```diff
@@ -0,0 +1,32 @@
+    color: #a5a5a5;
+}
+
+    .title__footer{
```
В этом файле могут быть только элементы блока `footer`, то есть `.footer__...`. А `.title__footer` — это элемент блока `.title`.

sergdenisov commented on less/footer.less:
```diff
@@ -0,0 +1,32 @@
+        color: #ffffff;
+    }
+
+    .main__footer{
```
Остальных тут и ниже это тоже касается.

sergdenisov commented on less/footer.less:
```diff
@@ -0,0 +1,32 @@
+}
+
+    .title__footer{
+        color: #ffffff;
```
`#fff`.

sergdenisov commented on less/footer.less:
```diff
@@ -0,0 +1,32 @@
+        padding-right: 60px;
+    }
+
+        .input__msg__footer{
```
Элемент элемента.

sergdenisov commented on less/header.less:
```diff
@@ -0,0 +1,20 @@
+    margin: 80px 0;
+}
+
+    .clients__header {
```
Здесь тоже самое с названиями элементов.

desile replied:
Это элементы header блоков clients, portfolio и т.п.

sergdenisov replied:
Опять же, стоит прочитать доку по БЭМу =).

desile replied:
https://ru.bem.info/methodology/html/#%D0%A0%D0%B0%D1%81%D0%BF%D0%BE%D0%BB%D0%BE%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5-%D0%B1%D0%BB%D0%BE%D0%BA%D0%B0-%D0%BE%D1%82%D0%BD%D0%BE%D1%81%D0%B8%D1%82%D0%B5%D0%BB%D1%8C%D0%BD%D0%BE-%D0%B4%D1%80%D1%83%D0%B3%D0%B8%D1%85-%D0%B1%D0%BB%D0%BE%D0%BA%D0%BE%D0%B2

Вроде вот там у них header и page__header

sergdenisov replied:
Там же написано, что это микс. То есть использование уже сделанных независимых блоков/элементов в HTML.

desile replied:
Да, вижу. И миксы входят в концепцию БЭМ. Так что, к сожалению, не могу понять, что не так.
header - подходит под понятие независимого блока, стили которого мы заимствуем для более специфичных элементов блоков.

sergdenisov replied:
В CSS в каждом файле должен быть только отдельный блок и его элементы. К примеру, в файле `header.less` у тебя лежит только этот блок `.header { ... }` и если бы у него были элементы, то `.header__element { ... }`. Но в данном случае их нет, поэтому только блок.
А `.clients__header { ... }` — это элемент блока `.clients { ... }`, и лежать он должен в файле `clients.less`.
А миксовать эти блоки/элементы ты можешь только при их использовании в HTML.

desile replied:
А, теперь понял :)
А вообще допускается, скажем в client.less, хранить, кроме самого блока client и его элементов, блоки вложенные в блок client?
Наша структура похожа на Flat https://ru.bem.info/methodology/filestructure/#flat - но я не нашел информации в которой однозначно говорится о вложенных блоках.

sergdenisov replied:
Нет, CSS не знает ничего о конкретных использованиях нескольких блоков вместе, они должны быть независимые. Блоки могут использовать в нескольких местах, в одном месте может быть микс, в другом может не быть.
https://ru.bem.info/methodology/css/#%D0%A0%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BA%D0%BE%D0%B4%D0%B0-%D0%BD%D0%B0-%D1%87%D0%B0%D1%81%D1%82%D0%B8

desile replied:
Понял, спасибо большое!

sergdenisov commented on less/img.less:
```diff
@@ -0,0 +1,90 @@
+  margin-bottom: 20px;
+}
+
+    .articles__person-card__img{
```
И здесь.

sergdenisov commented on less/img.less:
```diff
@@ -0,0 +1,90 @@
+
+    .articles__person-card__img{
+        display: inline-block;
+        width: @articles__avatar_size;
```
Нет импорта файла с переменными.

desile replied:
А разве не достаточно импорта файла переменных в all.less (seems it works) ?

sergdenisov replied:
Это будет работать, но это неявно и лучше так не делать.

sergdenisov commented on less/img.less:
```diff
@@ -0,0 +1,90 @@
+        background-size: cover;
+        width: 500px;
+        height: 314px;
+        margin: 0 0 0 0;
```
`margin: 0`.

sergdenisov commented on less/img.less:
```diff
@@ -0,0 +1,90 @@
+        margin: 0 0 0 0;
+    }
+
+    .news__row__block__img(){
```
Вот так совсем не надо делать, это еще и миксин! =)

sergdenisov commented on less/img.less:
```diff
@@ -0,0 +1,90 @@
+
+    .news__row__block__img(){
+        background-size: cover;
+        width: 480px;
```
Здесь не нужно хардкодить размер картинки, он должен зависеть от того, куда его вложили, то есть от сетки.

sergdenisov commented on less/img.less:
```diff
@@ -0,0 +1,90 @@
+            .news__row__block__img;
+        }
+
+    .team__member__img() {
```
Тоже самое.

sergdenisov commented on less/link.less:
```diff
@@ -8,3 +8,18 @@
+        display: block;
+    }
+
+    .link__credits__footer{
```
Элемент элемента.

sergdenisov commented on less/news.less:
```diff
@@ -0,0 +1,32 @@
+            align-items: center;
+        }
+
+            .news__other__block__valign{
```
Элемент элемент элемента))).

sergdenisov commented on less/news.less:
```diff
@@ -0,0 +1,32 @@
+        margin-bottom: 60px;
+    }
+
+        .news__row__block{
```
Тоже не очень говорящее название.

sergdenisov commented on less/slides.less:
```diff
@@ -0,0 +1,8 @@
+.slides{
```
Блок без стилей видимо не нужен.

sergdenisov commented on less/substrate.less:
```diff
@@ -1,3 +1,8 @@
     background: #efefef;
 }
+
+.dark_substrate {
```
`.substrate_dark`.

sergdenisov commented on less/substrate.less:
```diff
@@ -1,3 +1,8 @@
+
+.dark_substrate {
+    background: #14181d;
+    color: #f5f5f5;
```
По идее этот блок не должен задавать цвет текста.

desile replied:
А где правильнее задать цвет? В стиле конкретного раздела или завести отдельные стили для текста, которые будут применяться только в темных блоках?

sergdenisov replied:
Лучше отдельно для блока.

sergdenisov commented on less/tag.less:
```diff
@@ -0,0 +1,7 @@
+.tag{
```
Тоже самое.

sergdenisov commented on less/team.less:
```diff
@@ -0,0 +1,20 @@
+        margin-right: 50px;
+    }
+
+        .team__member__name{
```
И тут.

sergdenisov commented on less/text.less:
```diff
@@ -1,16 +1,35 @@
+        margin-bottom: 0;
+    }
+
+    .text_italic {
```
Нужно давать названия не по стилям, а по смыслу модификатора.

sergdenisov commented on less/underline.less:
```diff
@@ -0,0 +1,35 @@
+.underline {
+    border: none; /* Убираем границу для браузера Firefox */
```
Такие комменты не нужны.

desile replied:
Ссори, прорвалась копипаста)

sergdenisov commented on less/underline.less:
```diff
@@ -0,0 +1,35 @@
+
+    .portfolio_underline{
+        color: red; /* Цвет линии для остальных браузеров */
+        background-color: red; /* Цвет линии для браузера Firefox и Opera */
```
Не не, у нас не времена IE8, все должно работать одинаково во всех браузерах в данном случае, и все комменты не нужны =).

sergdenisov commented on less/variables.less:
```diff
@@ -1 +1,3 @@
 @color-text: #212121;
+@comment__img_size: 60px;
```
Переменные лучше называть не в стиле БЭМ.

desile replied:
Ок!

sergdenisov replied:
Не исправлено.

Commit:
using header style for its purpose

Commit:
replace tags for better semantic

Commit:
footer corrections:
 - vertical structure instead of horizontal
 - form and button instead of simple divs
 - placeholder style added with uppercase

Commit:
BEM corrections

Commit:
Row fixing and bem file structure

Commit:
Arrow absolute positioning

Commit:
Fixes of imports

sergdenisov commented on index.html:
```diff
@@ -4,7 +4,9 @@
         <link rel="stylesheet" href="css/bootstrap.min.css"/>
+        <link rel="stylesheet" href="css/font-awesome.min.css"/>
         <link rel="stylesheet/less" href="less/all.less"/>
+        <link rel="icon" href="data:,">
```
Это зачем?

sergdenisov commented on index.html:
```diff
@@ -4,7 +4,9 @@
         <title>Unicorn PSD Template</title>
         <meta charset="utf-8"/>
         <link rel="stylesheet" href="css/bootstrap.min.css"/>
+        <link rel="stylesheet" href="css/font-awesome.min.css"/>
```
Договаривались, что нужно без этого.

sergdenisov commented on index.html:
```diff
@@ -45,29 +47,29 @@ <h1 class="title">Storyteller</h1>
                 <div class="row justify-content-center">
                     <div class="col-10">
-                        <div class="intro">
+                        <div class="header whowhy_header">
```
`whowhy_header` — это модификатор, его можно использовать только с основным блоком.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
                 </div>
             </section>
+
+            <!--FEATURED ARTICLES-->
```
Про комментарии вроде говорил, что их быть не должно в таком виде.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <div class="container clients">
+                    <div class="row justify-content-center">
+                        <div class="col-10 justify-content-center">
+                        <div class="header clients__header">
```
Это обертка выглядит лишней.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <h2 class="title title_level-2">Clients</h2>
+                        </div>
+                        <ul>
+                            <li class="img clients__img"></li>
```
Контент должен лежать внутри `<li>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <li class="img clients__img"></li>
+                            <li class="img clients__img"></li>
+                            <li class="img clients__img"></li>
+                            <li class="fa fa-angle-right fa-4x clients__arrow"></li>
```
Стрелка — это не элемент списка, это `<button>` какой-нибудь, она должна лежать отдельно.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <li class="img clients__img"></li>
+                            <li class="fa fa-angle-right fa-4x clients__arrow"></li>
+                        </ul>
+                        </div>
```
Съехал весь блок на один отступ.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                    </div>
+                    <div class="underline portfolio__underline"></div>
+                    <ul class="row justify-content-center">
+                        <li class="portfolio__tag">Art</li>
```
Внутри элемента списка должна быть ссылка по логике.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </div>
+                    </div>
+                    <div class="underline portfolio__underline"></div>
```
Вроде писал, что это не должно лежать в HTML, если есть такая возможность. А тут она есть.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <li class="img portfolio__img"></li>
+                    </ul>
+                    <ul class="row justify-content-center">
+                        <li class="circle"></li>
```
Тут везде тоже должна внутри какая-нибудь кнопка лежать.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+            </section>
+
+            <!--CITATION-->
+            <section class="substrate_dark">
```
Уже писал про это — нельзя модификатор без блока.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <p class="text text_level-1">There's a special rung in hell reserved for people who waste good scotch.</p>
+                            </blockquote>
+                            <div class="col-8">
+                                <p class="text text_level-3 text_quote-author">Lt. Archie Hicox</p>
```
По семантике это не `<p>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <p class="text text_level-3 text_quote-author">Lt. Archie Hicox</p>
+                            </div>
+                        </div>
+                        <div class="row justify-content-center">
```
2 пустых блока, зачем?

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <div class="team">
+                    <div class="row justify-content-center">
+                        <div class="col-8">
+                            <div class="header">
```
Вроде тоже уже писал — блоки должны иметь осмысленные названия. Тут выходит, что у тебя в блоке "Заголовок" лежит заголовок и параграф с текстом.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                    <div class="row justify-content-center">
+                        <div class="team-member">
+                            <img src="img/member1.png" class="team-member__img"/>
+                            <div class="team-member__name">
```
Это должно быть заголовком типа `<h4>` — `<h6>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <div class="team-member__name">
+                                Paul Hindenburg
+                            </div>
+                            <div class="team-member__role">
```
И это тоже.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                    <div class="row">
+                                        <div class="col-6">
+                                            <div class="figures__number">100</div>
+                                            <div class="underline figures__underline"></div>
```
Тут тоже лучше через псевдо-элементы.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <div class="figures">
+                                    <div class="row">
+                                        <div class="col-6">
+                                            <div class="figures__number">100</div>
```
Это лучше заголовком тоже.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                        <div class="col-6">
+                                            <div class="figures__number">100</div>
+                                            <div class="underline figures__underline"></div>
+                                            <div class="title title_level-3">successful projects</div>
```
Это лучше заголовком тоже.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                        </div>
+                                    </div>
+                                </div>
+                                <div class="underline facts__underline"></div>
```
Это тоже.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </div>
+                        <div class="col-6">
+                                <img src="img/news1.jpg" class="news__img"/>
```
Съехало.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <div class="news">
+                    <div class="row justify-content-center news__row">
+                        <div class="col-6 news__article">
+                            <div>
```
Лишний элемент.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+            <section class="container">
+                <div class="news">
+                    <div class="row justify-content-center news__row">
+                        <div class="col-6 news__article">
```
Лучше `<article>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <div class="container">
+                    <div class="row justify-content-center other-news">
+                        <div class="col-4 other-news__preview">
+                            <div class="other-news__vline"></div>
```
Тоже лучше убрать.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                    <div class="row justify-content-center other-news">
+                        <div class="col-4 other-news__preview">
+                            <div class="other-news__vline"></div>
+                            <div>
```
Лишний.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+            <section class="substrate">
+                <div class="container">
+                    <div class="row justify-content-center other-news">
+                        <div class="col-4 other-news__preview">
```
Тоже лучше `<article>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <h6 class="text text_level-3">Travel / 21st March, 2014</h6>
+                            </div>
+                        </div>
+                        <div class="fa fa-angle-right fa-3x news__arrow"></div>
```
`<button>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+            <div class="container">
+                <div class="row footer__main">
+                    <div class="col-6">
+                        <p class="title title_level-3">UNICORN</p>
```
Заголовок.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <div class="row footer__main">
+                    <div class="col-6">
+                        <p class="title title_level-3">UNICORN</p>
+                        <form class="feedback" id="feedback">
```
`id` не нужно.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        </form>
+                    </div>
+                    <div class="col-2">
+                        <p class="title title_level-5">LINKS</p>
```
Заголовок.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            </div>
+                        </form>
+                    </div>
+                    <div class="col-2">
```
Тут везде, где списки ссылок, лучше `<nav>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                    <div class="col-2">
+                        <p class="title title_level-5">LINKS</p>
+                        <ul class="footer__menu">
+                            <li><a href="#" class="link footer__link">Home</a></li>
```
Каждый элемент с новой строки, в данном случае `<a>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        </ul>
+                    </div>
+                    <div class="col-2">
+                        <p class="title title_level-5">FRIENDS</p>
```
Заголовок.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        </ul>
+                    </div>
+                    <div class="col-2">
+                        <p class="title title_level-5">SOCIAL</p>
```
И тут.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                </div>
+                <div class="row">
+                    <div class="col-12">
+                        <button class="button feedback__button" form="feedback" type="submit">SEND ▶</button>
```
Это нужно положить в форму.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <div class="row">
+                    <div class="col-12">
+                        <button class="button feedback__button" form="feedback" type="submit">SEND ▶</button>
+                        <hr class="underline footer__underline">
```
`<hr>` тут не подходит.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                </div>
+                <div class="row">
+                    <div class="col-12 credits">
+                        <div>
```
Лишний.

sergdenisov commented on index.html:
```diff
@@ -110,8 +112,428 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <p class="text text_level-4">Chaos is found in greatest abundance wherever order is being sought.</p>
+                            <p class="text text_level-4">@2017 Awesome!</p>
+                        </div>
+                        <div>
```
`<nav>` — `<ul>` — `<li>`.

sergdenisov commented on less/articles.less:
```diff
@@ -0,0 +1,46 @@
+}
+
+
+    .person-card {
```
Мы это обсудили и ты сказал, что все понял. В файле `articles.less` может быть только блок `.articles` и его элементы. А этому блоку тут не место.

sergdenisov commented on less/button.less:
```diff
@@ -17,3 +13,17 @@
     }
 }
+
+    .promo__button {
```
Опять же, этому тут не место. Плюс так лучше не делать миксы, поскольку блок должен быть уникальный и независимый. А тут получается, что ты выносишь часть стилей в элемент.

sergdenisov commented on less/circle.less:
```diff
@@ -0,0 +1,12 @@
+
+    .circle_active {
+      background-color: #ffc107;
+    }
```
Пустая строка.

sergdenisov commented on less/clients.less:
```diff
@@ -0,0 +1,14 @@
+
+    .clients__header {
+        margin-bottom: 0;
+    }
```
Тоже.

sergdenisov commented on less/credits.less:
```diff
@@ -0,0 +1,5 @@
+    color: #fff;
+    display: flex;
+    justify-content: space-between;
+}
```
Тоже.

sergdenisov commented on less/features.less:
```diff
@@ -1,5 +1,6 @@
 .features {
-    padding: 75px 0 50px;
+    padding-top: 70px;
+    padding-bottom: 70px;
```
Лучше одной строкой писать по возможности.

sergdenisov commented on less/img.less:
```diff
@@ -0,0 +1,55 @@
+  margin-bottom: 20px;
+}
+
+    .person-card__img {
```
Тоже самое(.

sergdenisov commented on less/link.less:
```diff
@@ -8,3 +8,18 @@
 }
+
+
+    .footer__link {
```
((.

sergdenisov commented on less/news.less:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    .news__arrow {
+        margin-left: 560px;
```
Зачем?

sergdenisov commented on less/other-news.less:
```diff
@@ -0,0 +1,16 @@
+        align-items: center;
+    }
+
+    .other-news__vline {
```
Лучше без сокращений.

sergdenisov commented on less/substrate.less:
```diff
@@ -1,3 +1,8 @@
+
+.substrate_dark {
+    background: #14181d;
+    color: #f5f5f5;
```
Цвет отдельно лучше задавать.

sergdenisov commented on less/team.less:
```diff
@@ -0,0 +1,21 @@
+    padding-bottom: 70px;
+}
+
+.team-member{
```
В отдельный файл.

sergdenisov commented on less/team.less:
```diff
@@ -0,0 +1,21 @@
+    margin-right: 50px;
+}
+
+    .team-member__name{
```
Слиплось.

sergdenisov commented on less/text.less:
```diff
@@ -1,16 +1,36 @@
+        text-align: center;
+    }
+
+    .text_bold {
```
Выглядит как заголовок, а не текст.

sergdenisov commented on less/text.less:
```diff
@@ -1,16 +1,36 @@
     letter-spacing: 0.025em;
 }
 
+    .text_level-1 {
```
Видимо, нужно другое название.

sergdenisov commented on less/underline.less:
```diff
@@ -0,0 +1,37 @@
+    margin-right: auto;
+}
+
+    .portfolio__underline{
```
Тоже самое(.
